compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -1102,6 +1108,7 @@ struct InferBorrowKind<'a, 'tcx> {
     /// Place { V1, [ProjectionKind::Field(Index=1, Variant=0)] } : CaptureKind { E2, MutableBorrow }
     /// ```
     capture_information: InferredCaptureInformation<'tcx>,
+    fake_reads: Vec<(Place<'tcx>, FakeReadCause)>,
```

Comment:
Did you get a chance to try HashSet, because in case of 

```rust
let c = || {
   let (t1, _) = t;
   let (_, t2) = t;
}
```

we will introduce 2 fake reads, when one should suffice